pallets/stableswap/src/lib.rs

```diff
@@ -326,59 +342,94 @@ pub mod pallet {
 			Ok(())
 		}
 
-		/// Update given stableswap pool's parameters.
-		///
-		/// Updates one or more parameters of stablesswap pool ( amplification, trade fee, withdraw fee).
+		/// Update pool's fees.
 		///
-		/// If all parameters are none, `NothingToUpdate` error is returned.
+		/// Updates pool's trade fee and/or withdraw fee.
 		///
 		/// if pool does not exist, `PoolNotFound` is returned.
 		///
 		/// Parameters:
 		/// - `origin`: Must be T::AuthorityOrigin
 		/// - `pool_id`: pool to update
-		/// - `amplification`: new pool amplification or None
 		/// - `trade_fee`: new trade fee or None
 		/// - `withdraw_fee`: new withdraw fee or None
 		///
-		/// Emits `PoolUpdated` event if successful.
+		/// Emits `FeesUpdated` event if successful.
 		#[pallet::call_index(1)]
 		#[pallet::weight(<T as Config>::WeightInfo::update_pool())]
 		#[transactional]
-		pub fn update_pool(
+		pub fn update_pool_fees(
 			origin: OriginFor<T>,
 			pool_id: T::AssetId,
-			amplification: Option<u16>,
 			trade_fee: Option<Permill>,
 			withdraw_fee: Option<Permill>,
 		) -> DispatchResult {
 			T::AuthorityOrigin::ensure_origin(origin)?;
 
 			ensure!(
-				amplification.is_some() || trade_fee.is_some() || withdraw_fee.is_some(),
+				trade_fee.is_some() || withdraw_fee.is_some(),
 				Error::<T>::NothingToUpdate
 			);
 
 			Pools::<T>::try_mutate(pool_id, |maybe_pool| -> DispatchResult {
 				let mut pool = maybe_pool.as_mut().ok_or(Error::<T>::PoolNotFound)?;
 
-				pool.amplification = amplification.unwrap_or(pool.amplification);
-				ensure!(
-					T::AmplificationRange::get().contains(&pool.amplification),
-					Error::<T>::InvalidAmplification
-				);
 				pool.trade_fee = trade_fee.unwrap_or(pool.trade_fee);
 				pool.withdraw_fee = withdraw_fee.unwrap_or(pool.withdraw_fee);
-				Self::deposit_event(Event::PoolUpdated {
+				Self::deposit_event(Event::FeesUpdated {
 					pool_id,
-					amplification: pool.amplification,
 					trade_fee: pool.trade_fee,
 					withdraw_fee: pool.withdraw_fee,
 				});
 				Ok(())
 			})
 		}
 
+		/// Update pool's amplification.
+		///
+		/// Parameters:
+		/// - `origin`: Must be T::AuthorityOrigin
+		/// - `pool_id`: pool to update
+		/// - `future_amplification`: new desired pool amplification
+		/// - `future_timestamp`: future block number when the amplification is updated
+		///
+		/// Emits `AmplificationUpdated` event if successful.
+		#[pallet::call_index(2)]
+		#[pallet::weight(<T as Config>::WeightInfo::update_pool())]
+		#[transactional]
+		pub fn update_amplification(
+			origin: OriginFor<T>,
+			pool_id: T::AssetId,
+			future_amplification: u16,
+			future_timestamp: T::BlockNumber,
```

Comment:
Please stop referring to the block number as the timestamp. i recommend the same naming as I proposed for `AmplificationChanging` event. i would name it timestamp only in the case we really used timestamp